### ISSUE TITLE
Bootstrap the service using a node cluster

### ISSUE THREAD

**Issue by jmendiara** [Telefonica/alfalfa#3]
Currently, `startup.boostrap()` launchs the app managing only one process. Make all the `cluster` startup in a new `Startup.cluster()` method

**Comment by palmerabollo** [comment]
Why would you have two different ways to start you app if you can always do `startup.cluster`?

**Comment by jmendiara** [comment]
Because clustering, even with only one worker, behaves differently: Clustering means 1 master process (pid) and N worker processes. With only ONE cpu you will probably will switch executions contexts at OS level that can degrade perf.
Users may only want 1 pid to monitor (startup.bootstrap), which I think is the most common microservices scenario, without all the overhead for 2 processes. In older deployments with dedicated servers is where cluster may start adding value, where node can get as much as CPUs resources it can.

from the userland perspective, the proposal could  be (at `/bin/server --cluster`)
```js
if (process.args[2] === 'cluster') {
  startup.cluster();
} else {
  startup.bootstrap()
}```

this way, is up to the developer to decide enabling clustering mode.

What is your proposal? if the available CPUS is > 1 then start in cluster mode automatically?

**Comment by palmerabollo** [comment]
> What is your proposal? if the available CPUS is > 1 then start in cluster mode automatically?

Yes, to simplify the alfalfa learning curve (having two ways to start your app...). But I'd like to hear other people's opinion /cc @jlorgal @labajo @JoseAntonioRodriguez

**Comment by jmendiara** [comment]
I'm ok with that, but we must provide a way to avoid clustering startup regardless of the #CPUs: ie: to force a "just only one process" bootstrapping
